snappy-tools/sbin/collect-debug-artifacts.sh

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env bash
+
+#
+# Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you
+# may not use this file except in compliance with the License. You
+# may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License. See accompanying
+# LICENSE file.
+#
+
+#!/usr/bin/env bash
+
+usage="Usage: collect-debug-artifacts [ -c|--conf|--config conffile ] [ -h|--help ] \
+       [ -a|--all ] [ -v|--verbose ]"
+
+PWD=`pwd`
+while [ "$1" != "" ]; do
+key="$1"
+
+case $key in
+  -c|--conf|--config)
+  CONF_FILE="$2"
+  shift # past argument
+  ;;
+  -h|--help)
+  echo $usage
+  exit 0
+  ;;
+  -a|--all)
+  GET_EVERYTHING=1
+  ;;
+  -v|--verbose)
+  VERBOSE=1
+  ;;
+esac
+shift # past argument or value
+done
+
+# Check configurations and assign defaults
+function check_configs {
+  if [ -z "${CONF_FILE}" ]; then
+    CONF_FILE="../conf/debug.conf.template"
+  fi
+
+  if [ ! -f "${CONF_FILE}" ]; then
+    echo "Config file ${CONF_FILE} does not exists"
+    exit 1
+  fi
+
+  source $CONF_FILE
+
+  if [ ! -f "${MEMBERS_INFO_FILE}" ]; then
+    echo "MEMBERS_INFO_FILE file ${MEMBERS_INFO_FILE} does not exists"
+    exit 1
+  fi
+
+  if [ -z "${NO_OF_STACK_DUMPS}" ]; then
+    NO_OF_STACK_DUMPS=5
+  fi
+
+  if [ -z "${INTERVAL_BETWEEN_DUMPS}" ]; then
+    INTERVAL_BETWEEN_DUMPS=5
+  fi
+
+  if [ -z "${GET_EVERYTHING}" ]; then
+    GET_EVERYTHING=0
+  fi
+
+  if [ "${VERBOSE}" = "1" ]; then
+    echo CONF=$CONF_FILE
+    echo MEMINFO=$MEMBERS_INFO_FILE
+    echo NUM_STACK_DUMPS=$NO_OF_STACK_DUMPS
+    echo INTERVAL_BETWEEN_DUMPS=$INTERVAL_BETWEEN_DUMPS
+    echo GET_EVERYTHING=${GET_EVERYTHING}
+  fi
+}
+
+function collect_data {
+  host=$1
+  wd=$2
+  srv_num=$3
+
+  if [ "${VERBOSE}" = "1" ]; then
+    echo "Collecting data for process running on ${host} with working_dir ${wd}"
+  fi
+
+  # make a sub directory for this host and pid
+  out_dir="${PWD}/debug_data/${host}-${srv_num}"
+
+  collector_host="pnq-kneeraj3"
```

Comment:
No. This should be the collector host, on which the script is invoked. For testing I changed it to pnq-kneeraj. Will rectify it.